node.go

```diff
@@ -130,28 +142,24 @@ func (node *Node) Visit(opts *Options) (dirs, files int) {
 			vpaths: node.vpaths,
 		}
 		d, f := nnode.Visit(opts)
+		// "prune" option, hide empty directories
+		if nnode.IsDir() && opts.Prune && f == 0 {
+			continue
+		} else if nnode.IsDir() && opts.MatchDirs && opts.IPattern != "" && nnode.match(opts.IPattern, opts) {
+			continue
+		}
 		if nnode.err == nil && !nnode.IsDir() {
 			// "dirs only" option
 			if opts.DirsOnly {
 				continue
 			}
-			var rePrefix string
-			if opts.IgnoreCase {
-				rePrefix = "(?i)"
-			}
 			// Pattern matching
-			if opts.Pattern != "" {
-				re, err := regexp.Compile(rePrefix + opts.Pattern)
-				if err == nil && !re.MatchString(name) {
-					continue
-				}
+			if !dirMatch && opts.Pattern != "" && !nnode.match(opts.Pattern, opts) {
+				continue
```

Comment:
```suggestion
			// If dir matched, the pattern checks is diabled for its content.
			if !dirMatch && opts.Pattern != "" && !nnode.match(opts.Pattern, opts) {
				continue
```

Added some docs for future readers.

Reply:
I am confused, because it looks like you want to add a comment which describes a scenario which doesn't apply to the existing line of code?